lib/storage/client.js

```diff
@@ -1196,6 +1196,42 @@ class Client extends CryptoConsumer {
     }
     return returnValues
   }
+  /**
+   *
+   * @param {string} clientID   UUID assigned by TozStore, used to identify the client; optional.
+   * @param {String} groupName Group name you want to get information regarding.
+   *
+   * @return {Object}  The group object or null if it does not exist.
+   */
+  async groupInfo(clientID = null, groupName) {
+    const urlData = {
+      client_id: clientID,
+    }
+    let encodedUrl = urlEncodeData(urlData)
+    let current = 'group_names=' + groupName
+    encodedUrl += '&' + current
+
+    let response = await this.authenticator.tsv1Fetch(
+      `${this.config.apiUrl}/v2/storage/groups?${encodedUrl}`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+    )
+    let groupsJson = await validateResponseAsJSON(response)
+    let finalgroup
+    groupsJson.groups.forEach(group => {
```

Comment:
`find()` will be cleaner than `foreach()` (plus is more efficient, though it's a total micro optimization that doesn't really matter much): 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

Might also consider checking to make sure the group was actually found and erroring on the not-found case.

```js
const finalGroup = groupsJson.groups.find(g => g.group_name == groupName)
if (!finalGroup) {
  // handle error
}
return {
  group: Group.decode(finalGroup)
}
```

Reply:
Also - what is the value of the object wrapping the returned group? I think in this instance it should just return the `Group` type directly unless I am missing something?